volgen/core.go

```diff
@@ -14,59 +15,84 @@ const (
 )
 
 // GenerateVolfile function will do all task from graph generation to volfile generation
-func GenerateVolfile(vinfo *volume.Volinfo) {
+func GenerateVolfile(vinfo *volume.Volinfo) error {
 	var cpath string
+	var err error
+	var f *os.File
+
 	graph := GenerateGraph(vinfo, "CLIENT")
 
 	//Generate client volfile
-	getClientFilePath(vinfo, &cpath)
-	f, err := os.Create(cpath)
+	err = getClientFilePath(vinfo, &cpath)
+	if err != nil {
+		return err
+	}
+
+	f, err = os.Create(cpath)
 	if err != nil {
-		panic(err)
+		return err
 	}
-	defer f.Close()
 	graph.DumpGraph(f)
+	f.Close()
 
 	//Generate volfile for server
-	for _, v := range vinfo.Bricks {
+	for _, b := range vinfo.Bricks {
 		sgraph := GenerateGraph(vinfo, "SERVER")
 
 		//Generate volfile for server
-		getServerFilePath(vinfo, &cpath, v)
-		f, err := os.Create(cpath)
+		err = getServerFilePath(vinfo, &cpath, b)
+		if err != nil {
+			return err
+		}
+
+		f, err = os.Create(cpath)
 		if err != nil {
-			panic(err)
+			return err
 		}
-		defer f.Close()
+
 		sgraph.DumpGraph(f)
+		f.Close()
 	}
+	return nil
 }
 
 func getVolumeDir(vinfo *volume.Volinfo, dir *string) {
 	*dir = fmt.Sprintf("%s/vols/%s", workdir, vinfo.Name)
 }
 
-func getServerFilePath(vinfo *volume.Volinfo, path *string, brick string) {
+func getServerFilePath(vinfo *volume.Volinfo, path *string, brickinfo volume.Brickinfo) error {
 	var vdir string
 	getVolumeDir(vinfo, &vdir)
 
-	hname, _ := os.Hostname()
+	//TODO : take the hostname from brickinfo
+	hname := brickinfo.Hostname
 	// Create volume directory (/var/lib/glusterd/vols/<VOLNAME>)
 	err := os.MkdirAll(vdir, 0666)
 	if err != nil {
-		panic(err)
+		return err
+	}
+
+	slash_to_hyphen := func(r rune) rune {
```

Comment:
The `filepath` package has a function that does this already, [`filepath.FromSlash()`](https://golang.org/src/path/filepath/path.go?s=4601:4635#L161) . Use this instead.